src/Agent.Worker/Release/Artifacts/BuildArtifact.cs

```diff
@@ -236,8 +236,20 @@ private bool Match(ServerBuildArtifact buildArtifact, ArtifactDefinition artifac
                     DownloadBufferSize = executionContext.Variables.Release_Download_BufferSize ?? ContainerFetchEngineDefaultOptions.DownloadBufferSize
                 };
 
+                int fetchEngineTimeoutSeconds;
+                if (!int.TryParse(Environment.GetEnvironmentVariable("VSTS_HTTP_TIMEOUT") ?? string.Empty, out fetchEngineTimeoutSeconds))
```

Comment:
don't see why we should cap the max. timeout. instead it is better to not let the file download to timeout. there is already a timeout for the deployment configured and if download exceeds this timeout, it would be canceled by agent.

@stephenmichaelf - is it fine if you let @omeshp to take this fix forward?